neo/Network/LocalNode.cs

```diff
@@ -206,20 +206,33 @@ private void Blockchain_PersistCompleted(object sender, Block block)
             Transaction[] remain;
             lock (mem_pool)
             {
+                // Remove the transactions that made it into the block
                 foreach (Transaction tx in block.Transactions)
-                {
                     mem_pool.Remove(tx.Hash);
-                }
                 if (mem_pool.Count == 0) return;
 
                 remain = mem_pool.Values.ToArray();
                 mem_pool.Clear();
+                
+                ConcurrentBag<Transaction> verified = new ConcurrentBag<Transaction>();
+                // Reverify the remaining transactions in the mem_pool
+                remain.AsParallel().ForAll(tx =>
+                {
+                    if (tx.Verify(remain))
+                        verified.Add(tx);
```

Comment:
Re-verifying the transactions in `PersistCompleted` can cause the thread to block if there is a large amount of transactions in the memory pool at this time.

Reply:
Explain why that is a problem. Isn't it more important than other things at the time that a block just gets persisted?

Reply:
Ah, actually that would negate some of what I did to improve the performance when syncing the chain. I think it only makes sense to re-verify them immediately if it has been at least 10 seconds since the last block; otherwise put them in the temp_pool. What do you think of that? That would allow it to bypass in the case that the chain is syncing and it is getting blocks quickly.

Reply:
In the case that the chain is fully synced though, it seems like it is better behavior to re-verify the transactions after the block rather than have it wait. I thought of another idea actually. What if it were to just verify the first 1000 or so transactions with the highest fees and put the rest in the temp_pool (and if the last block had been less than 10 seconds ago, then it just puts them all in the temp_pool).

Reply:
I went ahead and added a check so that when syncing the chain it won't re-verify the transactions if there is a large amount in the memory pool at the time.